java/runtime/src/main/java/org/ray/runtime/runner/RunManager.java

```diff
@@ -146,23 +147,33 @@ public void startRayProcesses(boolean isHead) {
 
   private void startRedisServer() {
     // start primary redis
-    String primary = startRedisInstance(rayConfig.nodeIp, rayConfig.headRedisPort, null);
+    String primary = startRedisInstance(rayConfig.nodeIp,
+        rayConfig.headRedisPort, rayConfig.headRedisPassword, null);
     rayConfig.setRedisAddress(primary);
     try (Jedis client = new Jedis("127.0.0.1", rayConfig.headRedisPort)) {
+      if (!StringUtil.isNullOrEmpty(rayConfig.headRedisPassword)) {
+        client.auth(rayConfig.headRedisPassword);
+      }
       client.set("UseRaylet", "1");
       // Register the number of Redis shards in the primary shard, so that clients
       // know how many redis shards to expect under RedisShards.
       client.set("NumRedisShards", Integer.toString(rayConfig.numberRedisShards));
 
       // start redis shards
       for (int i = 0; i < rayConfig.numberRedisShards; i++) {
-        String shard = startRedisInstance(rayConfig.nodeIp, rayConfig.headRedisPort + i + 1, i);
+        String shard = startRedisInstance(rayConfig.nodeIp,
+            rayConfig.headRedisPort + i + 1, rayConfig.headRedisPassword, i);
         client.rpush("RedisShards", shard);
       }
     }
   }
 
-  private String startRedisInstance(String ip, int port, Integer shard) {
+  private String startRedisInstance(String ip, int port, String password, Integer shard) {
+    String passwordCommand = "";
+    if (!StringUtil.isNullOrEmpty(password)) {
```

Comment:
@raulchen  I couldn't find some thing like `Lists.of()` to replace `ImmutableLists.of()` here.
Could you plz give me some suggestions?

Reply:
You can use `Lists.newArrayList`